packages/flutter_tools/lib/runner.dart

```diff
@@ -75,6 +75,15 @@ Future<int> _handleToolError(
     bool reportCrashes,
     String getFlutterVersion(),
     ) async {
+  // Wait for any existing engine crash reports to finish.
+  if (CrashReportSender.engineCrash != null && CrashReportSender.engineCrash.reportStatus != null) {
```

Comment:
I didn't see how it could end up with an existing and unfinished engine crash report. It seems that we always `await CrashReportSender.instance.checkAndSendEngineReport`. If we don't have to wait for existing reports, we can get rid of `engineCrash` field its `reportStatus`.